onadata/apps/api/viewsets/floip_viewset.py

```diff
@@ -118,3 +120,13 @@ def responses(self, request, uuid=None):
             data['attributes']['responses'] = floip_list(queryset)
 
         return Response(data, headers=headers, status=status_code)
+
+
+def floip_responses_filter(query_params, queryset):
```

Comment:
Might be a bit picky but could we override the [`filter_queryset`](https://www.django-rest-framework.org/api-guide/filtering/#example) function here, to handle the necessary filtering for this particular viewset ? It would allow one to access the request object through `self.request` and may make it easier to add more filters we may want in the future.